autoPyTorch/pipeline/components/setup/traditional_ml/traditional_learner/base_traditional_learner.py

```diff
@@ -0,0 +1,264 @@
+import json
+import logging.handlers
+import os as os
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+from catboost import CatBoost
+
+import numpy as np
+
+import pandas as pd
+
+from sklearn.base import BaseEstimator
+from sklearn.utils import check_random_state
+
+from autoPyTorch.constants import REGRESSION_TASKS, STRING_TO_TASK_TYPES
+from autoPyTorch.pipeline.components.training.metrics.utils import get_metrics
+from autoPyTorch.utils.logging_ import get_named_client_logger
+
+
+class BaseTraditionalLearner:
+    """
+    Base wrapper class for Traditional Learners.
+
+    Args:
+        task_type (str):
+            Type of the current task. Currently only tabular
+            tasks are supported. For more info on the tasks
+            available in AutoPyTorch, see
+            `autoPyTorch/constants.py`
+        output_type (str):
+            Type of output. The string depends on the output of
+            sklearn's type_of_target. `see
+            <https://scikit-learn.org/stable/modules/generated/sklearn.utils.multiclass.type_of_target.html>`
+        logger_port (int) (default=logging.handlers.DEFAULT_TCP_LOGGING_PORT)
+        random_state (Optional[np.random.RandomState], default=None):
+        name (str, default=''):
+            Name of the learner, when not specified,
+            uses the name of the class
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 output_type: str,
+                 logger_port: int = logging.handlers.DEFAULT_TCP_LOGGING_PORT,
+                 random_state: Optional[np.random.RandomState] = None,
+                 name: str = ''):
```

Comment:
do you think it makes sense to use None? like having an `Optional[str] = None`?

Reply:
yes we should avoid mutable types in the init